src/types/types.ts

```diff
@@ -2591,6 +2591,93 @@ export interface DebugPrestate {
   storage: Record<string, string>;
 }
 
+/*
+ * SDK Representation of a user operation and bundler types
+ */
+export interface UserOperation {
+  sender: string;
+  nonce: string;
+  initCode: string;
+  callData: string;
+  callGasLimit: number;
+  verificationGasLimit: number;
+  preVerificationGas: number;
+  maxFeePerGas: number;
+  maxPriorityFeePerGas: number;
+  paymasterAndData: string;
+  signature: string;
+  entryPoint: string;
+  blockNumber: number;
+  blockHash: string;
+  transactionHash: string;
+}
+
+export interface UserOperationRequest {
+  sender: string;
+  nonce: string;
+  initCode: string;
+  callData: string;
+  callGasLimit: number;
+  verificationGasLimit: number;
+  preVerificationGas: number;
+  maxFeePerGas: number;
+  maxPriorityFeePerGas: number;
+  paymasterAndData: string;
+  signature: string;
+}
+
+export interface UserOperationReceipt {
+  userOpHash: string;
+  entryPoint: string;
+  sender: string;
+  nonce: number;
+  paymaster: string;
+  actualGasCost: number;
+  actualGasUsed: number;
+  success: boolean;
+  reason: string;
+  logs: string[];
+  receipt: UserOperationReceiptObject;
+}
+
+export interface UserOperationReceiptLog {
+  blockHash: string;
+  blockNumber: string;
+  transactionIndex: string;
+  address: string;
+  logIndex: string;
+  data: string;
+  removed: boolean;
+  topics: string[];
+  transactionHash: string;
+  logsBloom: string;
+  root: string;
+  status: number;
+  effectiveGasPrice: string;
+}
+
+export interface UserOperationReceiptObject {
+  blockHash: string;
+  blockNumber: string;
+  transactionIndex: string;
+  transactionHash: string;
+  from: string;
+  to: string;
+  cumulativeGasUsed: string;
+  gasUsed: string;
+  contractAddress: string;
+  logs: UserOperationReceiptLog[];
+}
+
+export interface UserOperationEstiamateGasResponse {
+  /* Gas overhead of this UserOperation */
+  preVerificationGas: number;
```

Comment:
do gas fields have a chance of overflowing on different chains? i'm leaning that we standardize to returning the response as an integer string and casting to BigNumber (to be replaced by BigInt)

Reply:
+1, these should be `bigint` types